util/retrieve-cluster-logs.py

```diff
@@ -0,0 +1,213 @@
+import gzip
+import json
+import logging
+import os
+import sys
+import tarfile
+import tempfile
+from datetime import datetime
+
+import argparse
+import boto3
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__file__)
+
+
+def err_and_exit(message):
+    """Log the given error message and exit nonzero."""
+    LOGGER.error(message)
+    sys.exit(1)
+
+
+def get_default_bucket_prefix(cluster_name):
+    """Get default keypath to store logs under in an s3 bucket for the given cluster."""
+    return "{cluster_name}-logs-{end_time}".format(
+        cluster_name=cluster_name, end_time=datetime.now().strftime("%Y%m%d%H%M%S")
+    )
+
+
+def get_log_group(args):
+    """Get log group for args.cluster."""
+    log_group_name = "/aws/parallelcluster/{}".format(args.cluster)
+    logs = boto3.client("logs", region_name=args.region)
+    paginator = logs.get_paginator("describe_log_groups")
+    for result in paginator.paginate(logGroupNamePrefix=log_group_name):
+        for group in result.get("logGroups"):
+            if group.get("logGroupName") == log_group_name:
+                return group
+    err_and_exit(
+        "Unable to find log group for cluster {cluster}. Expected to find one named {log_group_name}".format(
+            cluster=args.cluster, log_group_name=log_group_name
+        )
+    )
+
+
+def verify_bucket_exists_in_region(args):
```

Comment:
minor: as internal convention we are using one underscore prefix for functions that are only used in the module itself.

Reply:
Since this is meant to be used as a script, that would mean all of the functions should have that prefix, correct?

Reply:
If the script remains isolated from the rest of the code it is ok for me to keep the code as is.
If we'll move this functionality in our cli I'd add the underscore prefix.

Reply:
Agreed. Since I'm checking this in as a standalone script I'll resolve this, but I'll be sure to adhere to this convention if adding this code to the CLI package.